Assets/HoloToolkit/Utilities/Editor/BuildCommands.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Implements functionality for building HoloLens applications
+    /// </summary>
+    public static class BuildCommands
+    {
+        public static readonly string BuildLocation = "Builds/HoloLens";
+
+        /// <summary>
+        /// Do a build configured for the HoloLens, returns the error from BuildPipeline.BuildPlayer
+        /// </summary>
+        public static string BuildForHololens()
+        {
+            var scenes = EditorBuildSettings.scenes.Where(s => s.enabled).Select(s => s.path);
+
+            // Cache the current settings
+            BuildTarget oldBuildTarget = EditorUserBuildSettings.activeBuildTarget;
+            WSASDK oldSDK = EditorUserBuildSettings.wsaSDK;
+            WSAUWPBuildType oldBuildType = EditorUserBuildSettings.wsaUWPBuildType;
+
+            // Set the sdk and build type
+            EditorUserBuildSettings.wsaSDK = WSASDK.UWP;
+            EditorUserBuildSettings.wsaUWPBuildType = WSAUWPBuildType.D3D;
+
+            string error = BuildPipeline.BuildPlayer(scenes.ToArray(), BuildLocation, BuildTarget.WSAPlayer, BuildOptions.None);
+
+            // Restore the sdk and build type
+            EditorUserBuildSettings.wsaSDK = oldSDK;
+            EditorUserBuildSettings.wsaUWPBuildType = oldBuildType;
+            EditorUserBuildSettings.SwitchActiveBuildTarget(oldBuildTarget);
+
+            return error;
```

Comment:
If executed from the menu this logs to the Unity console. If executed from the command line users can direct the logs to any file they desire via Unity's -logFile command. See here: https://docs.unity3d.com/Manual/CommandLineArguments.html